From ce33360524307b6da4f996a9f465260e121011f8 Mon Sep 17 00:00:00 2001
From: Vincent Ambo <tazjin@google.com>
Date: Tue, 11 Feb 2020 00:40:37 +0000
Subject: [PATCH] fix(control): Ensure control client has a logger function

For some reason this field ended up being nil in taillogin, which made
it difficult to actually log in.
---
 control/controlclient/auto.go | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/control/controlclient/auto.go b/control/controlclient/auto.go
index 67f187f..09cdf6f 100644
--- a/control/controlclient/auto.go
+++ b/control/controlclient/auto.go
@@ -160,6 +160,9 @@ func NewNoStart(opts Options) (*Client, error) {
 //
 // It should only be called for clients created by NewNoStart.
 func (c *Client) Start() {
+	c.logf = func(f string, args ...interface{}) {
+		fmt.Printf(f, args)
+	}
 	go c.authRoutine()
 	go c.mapRoutine()
 }
-- 
2.23.0

